spec/models/validations/sales/sale_information_validations_spec.rb

```diff
@@ -102,10 +102,62 @@
       let(:record) { build(:sales_log, hodate: Time.zone.parse("2023-07-01"), saledate: Time.zone.parse("2023-07-01")) }
 
       it "does not add an error" do
-        sale_information_validator.validate_pratical_completion_date_before_saledate(record)
+        sale_information_validator.validate_practical_completion_date_before_saledate(record)
 
         expect(record.errors[:hodate]).to be_present
       end
     end
   end
+
+  describe "#validate_years_living_in_property_before_purchase" do
+    context "when proplen blank" do
+      let(:record) { build(:sales_log, proplen: nil) }
+
+      it "does not add an error" do
+        sale_information_validator.validate_years_living_in_property_before_purchase(record)
+
+        expect(record.errors).not_to be_present
+      end
+    end
+
+    context "when type blank" do
+      let(:record) { build(:sales_log, type: nil) }
+
+      it "does not add an error" do
+        sale_information_validator.validate_years_living_in_property_before_purchase(record)
+
+        expect(record.errors).not_to be_present
+      end
+    end
+
+    context "when proplen 0" do
+      let(:record) { build(:sales_log, proplen: 0) }
+
+      it "does not add an error" do
+        sale_information_validator.validate_years_living_in_property_before_purchase(record)
+
+        expect(record.errors).not_to be_present
+      end
+    end
+
+    context "when type Rent to Buy and proplen > 0" do
+      let(:record) { build(:sales_log, proplen: 1, type: 28) }
+
+      it "adds an error" do
+        sale_information_validator.validate_years_living_in_property_before_purchase(record)
+
+        expect(record.errors).to be_present
```

Comment:
Could we check for the specific error message?
` expect(record.errors[:type]).to include(expected_error_message)` etc